Content.Server/Abilities/Boxer/BoxingSystem.cs

```diff
@@ -0,0 +1,45 @@
+using Content.Server.Weapon.Melee;
+using Content.Server.Stunnable;
+using Content.Shared.Inventory.Events;
+using Content.Server.Weapon.Melee.Components;
+using Content.Server.Clothing.Components;
+using Content.Server.Damage.Components;
+using Content.Server.Damage.Events;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+using Robust.Shared.Containers;
+
+namespace Content.Server.Abilities.Boxer
+{
+    public sealed class BoxingSystem : EntitySystem
+    {
+        [Dependency] private readonly StunSystem _stunSystem = default!;
+        [Dependency] private readonly IRobustRandom _robustRandom = default!;
+        [Dependency] private readonly SharedContainerSystem _containerSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<BoxerComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<BoxerComponent, MeleeHitEvent>(ApplyBoxerModifiers);
+            SubscribeLocalEvent<BoxingGlovesComponent, StaminaMeleeHitEvent>(OnStamHit);
+        }
+
+        private void OnInit(EntityUid uid, BoxerComponent boxer, ComponentInit args)
+        {
+            var meleeComp = EnsureComp<MeleeWeaponComponent>(uid);
+            meleeComp.Range *= boxer.RangeBonus;
```

Comment:
Should be a trycomp, see https://github.com/space-wizards/space-station-14/issues/8423